src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,9 +259,31 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             }
 
             // variables declared using var cannot be used further in the same initialization expression.
-            if (initializer.DescendantNodesAndSelf()
-                    .Where(n => (n as IdentifierNameSyntax)?.Identifier.ValueText.Equals(identifier.ValueText) == true)
-                    .Any(n => semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true))
+            bool isReusedInInitializer(SyntaxNode n)
+            {
+                // naming collision prerequisite
+                if (!(n is IdentifierNameSyntax id
+                    && id.Identifier.ValueText.Equals(identifier.ValueText)))
+                {
+                    return false;
+                }
+
+                // case of variable direct use: int x = x * 2;
+                if (semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true)
+                {
+                    return true;
+                }
```

Comment:
💡 Include a blank line after the closing `}` here and at the end of the local function.